scripts/search.py

```diff
@@ -131,23 +131,22 @@ def filter_forward_pairs_by_reverse_pairs(forward_blast_high_scoring_pairs, reve
     filterable_forward_blast_results = list(forward_blast_high_scoring_pairs)
     print(">> Checking if forward hit subjects have better reciprocal hits than query.")
     for hit in forward_blast_high_scoring_pairs:
-        query_protein = blast_graph.getVertex(hit[0])
-        subject_protein = blast_graph.getVertex(hit[1])
+        subject_protein = blast_graph[hit[1]]
 
-        top_back_hit_score = 0
         # Find the top score of the best reciprocal BLAST hit
-        for backHit in subject_protein.getConnections():
-            back_hit_score = subject_protein.getWeight(
-                backHit)  # The edge weight between the subject and its reciprocal BLAST hit is the BLAST score
+        top_back_hit_score = 0
+        for back_hit_id in subject_protein:
+            back_hit_score = get_edge_attribute(subject_protein, back_hit_id,
+                                                'bitscore-rev', none_value=-1)
             if back_hit_score >= top_back_hit_score:
                 top_back_hit_score = back_hit_score
 
         # Check if the query is the best reciprocal BLAST hit for the subject
         delete_hit = False
-        if query_protein in subject_protein.getConnections():
+        if hit[0] in subject_protein:
             # The edge weight between the subject and the query is the reciprocal BLAST score
-            back_hit_to_query_score = subject_protein.getWeight(query_protein)
-
+            back_hit_to_query_score = get_edge_attribute(subject_protein, hit[0],
+                                                         'bitscore-rev', none_value=-1)
```

Comment:
@jmtsuji Double-check that the none_value=-1 parameter is correct and doesn't cause any issues.

Reply:
@LeeBergstrand I double checked the logic here, and the -1 works. Setting this value to -1 (rather than 0) prevents a rare edge case where a nonexistent reverse hit might appear like it actually exists with bitscore=0. I've added a comment to the code to clarify why this choice was made. How does this look to you?

Reply:
Sounds good!